content/docs/scaffolder/scaffolder-parameters/index.md

```diff
@@ -0,0 +1,563 @@
+
+```yaml
+properties:
+  names:
+    title: Names
+    type: array
+    ui:widget: "CheckboxesWidget"
+    uniqueItems: true
+    items:
+      enum: 
+        - Tom
+        - Bob
+```
+
+##### Enum Display Names
+
+You can render a different name for your enum values using `enumNames`:
+
+```yaml
+properties:
+  city:
+    title: City
+    type: string
+    enumNames: 
+      - New York
+      - Amsterdam
+      - Hong Kong
+    enum:
+      - nyc
+      - ams
+      - hk
+```
+
+This is particularly useful for encoding additional hidden data in your enum values that can be used in later template steps:
+
+```yaml
+properties:
+  city:
+    title: City
+    type: string
+    enumNames: 
+      - New York
+      - Amsterdam
+      - Hong Kong
+    enum:
+      - name: 'New York'
+        lat: 40
+        lon: 74
+      - name: Amsterdam
+        lat: 52
+        lon: 5
+      - name: 'Hong Kong'
+        lat: 22
+        lon: 114
+
+steps:
+  - id: log-message
+    name: Log Message
+    action: debug:log
+    input:
+      message: 'The city ${{ parameters.city.name }} is at ${{ parameters.city.lat }},${{ parameters.city.lon }}'
+```
+
+### Field Validation
+
+You can use [react-jsonschema-form](https://rjsf-team.github.io/react-jsonschema-form/) to perform validation on input fields using a regex in the `pattern` filed or character counts using `maxLength` and `minLength`.
+
+```yaml
+parameters:
+  properties:
+    name:
+      title: Simple text input
+      type: string
+      description: Description about input
+      maxLength: 8
+      pattern: '^([a-zA-Z][a-zA-Z0-9]*)(-[a-zA-Z0-9]+)*$'
+      ui:autofocus: true
+      ui:help: 'Hint: additional description...'
+    errorMessage:
+      properties:
+        name: '1-8 alphanumeric tokens (first starts with letter) delimited by -'
+```
+
+
+### Multi Page Forms
+
+You can choose to break up the parameters into form pages or collect all the parameters in one single page.
+
+**Single Page:**
+```yaml
+parameters:
+  properties:
+    name:
+      title: Name
+      type: string
+    email:
+      title: Email
+      type: string
+      maxLength: 8
+      pattern: '^[^\s@]+@([^\s@.,]+\.)+[^\s@.,]{2,}$'
+    role:
+      title: Role
+      type: string
+    company:
+      title: Company Name
+      type: string
+```
+
+**Multi Page:**
+```yaml
+parameters:
+  - title: Provide some info about yourself
+    required:
+      - name
+      - email
+    properties:
+      name:
+        title: Name
+        type: string
+      email:
+        title: Email
+        type: string
+        maxLength: 8
+        pattern: '^[^\s@]+@([^\s@.,]+\.)+[^\s@.,]{2,}$'
+  - title: Provide some info about your occupation
+    properties:
+      role:
+        title: Role
+        type: string
+      company:
+        title: Company Name
+        type: string
+```
+
+In a multipage form, each parameter page must contain `title` and `properties`
+
+
+### Complex Form Fields
```

Comment:
Maybe in this section a reference to custom plugins and adding custom form fields via that?

Reply:
as in, link to the bottom of this page